includes/Core/Authentication/Authentication.php

```diff
@@ -443,6 +451,47 @@ public function get_disconnect_url() {
 		);
 	}
 
+	/**
+	 * Gets the proxy permission url
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string Proxy permission url.
+	 */
+	private function get_proxy_permissions_url() {
+		return add_query_arg(
+			array(
+				'action' => Google_Proxy::ACTION_PERMISSIONS,
+				'nonce'  => wp_create_nonce( Google_Proxy::ACTION_PERMISSIONS ),
+			),
+			admin_url( 'index.php' )
+		);
+	}
+
+	/**
+	 * Handles proxy permissions
+	 *
+	 * @since n.e.x.t
+	 */
+	private function handle_proxy_permissions() {
+		if ( ! current_user_can( Permissions::AUTHENTICATE ) ) {
+			wp_die( esc_html__( 'You need a higher level of permission.', 'google-site-kit' ) );
+		}
+
+		if ( ! $this->credentials->using_proxy() ) {
+			wp_die( esc_html__( 'You need a higher level of permission.', 'google-site-kit' ) );
```

Comment:
This is not an accurate message, there is no permission issue here. There is already a similar message for the same condition in the `handle_sync_site_fields` method which we can reuse: _Site Kit is not configured to use the authentication proxy._